### DISCUSSION TITLE
Grove Basic Kit for Raspberry Pi Pico | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#1642] [Q&A]
# Grove Basic Kit for Raspberry Pi Pico | Seeed Studio Wiki

Grove Shield / Grove Starter Kit for Raspberry Pi Pico

https://wiki.seeedstudio.com/Grove-Starter-Kit-for-Raspberry-Pi-Pico/

**Comment by BGilmore21** [comment]
This demo assumes that Micropython firmware is already installed on Raspberry Pi Pico.  Instructions for installing this firmware are at https://projects.raspberrypi.org/en/projects/getting-started-with-the-pico/3

**Comment by BGilmore21** [comment]
The link to lcd1602.py is dead.  There is a simple micropython driver for the LCD display at github user Bhavithiran97's page, but using it requires recoding the demo.

**Comment by paul0307** [comment]
Project 4: Flashing Colors - First of all sorry for being new to all this but I get the errors below after copying and paste the code for Flashing Colors any help please as I'm still learning :-)

>>> %Run -c $EDITOR_CONTENT

MPY: soft reboot
Traceback (most recent call last):
  File "<stdin>", line 15, in <module>
  File "ws2812.py", line 45, in __init__
AttributeError: type object 'SPI' has no attribute 'MASTER'
>>>

**Reply by limengdu** [reply]
Hello, we are concerned about the problem you mentioned with this wiki, next we will need a little time to test the code of this programme and get back to you, please wait. @Carla-Guo Please help test the programme.

**Reply by Carla-Guo** [reply]
> Project 4: Flashing Colors - First of all sorry for being new to all this but I get the errors below after copying and paste the code for Flashing Colors any help please as I'm still learning :-)
> 
> > > > %Run -c $EDITOR_CONTENT
> 
> MPY: soft reboot Traceback (most recent call last): File "", line 15, in File "ws2812.py", line 45, in **init** AttributeError: type object 'SPI' has no attribute 'MASTER'
> 
> > > >

@paul0307 
Hi~ I just tested the demo on my side, and it works well. 👌
Could you double-check if you’ve downloaded correct ws2812.py files? 
The link is here: https://files.seeedstudio.com/wiki/Grove-Temperature-Humidity-Sensor/ws2812.py 👈
And if possible, could you also share the steps you took so I can try to reproduce it exactly?

**Comment by paul0307** [comment]
Hi,

I nuked my Pico and downloaded the file again, Also copied and pasted the code from the WIKI and it worked.

Now I know I’m new to this but I’m sure I did it correct the first time so I’m not sure what is going on.

Thanks for your time in looking into this for me now onto the next part of the kit :-)

Kind Regards 

Paul

**Comment by thegreatgunbantoad** [comment]
For Project 6, I don't think the servo should be being run through 0 to 65535, but rather the 2.5% and 12.5% value of that range.  It seems the servo is a 50Hz, 0 to 180 degree range where 2.5% is one end and 12.5% the other.  The Servo specs dont really tell you much.

**Reply by nfs0619** [reply]
Thank you for your feedback. We have received your message and are currently verifying this issue. If you do not receive further updates, please feel free to reach out to our technical support at [techsupport@seeed.io](mailto:techsupport@seeed.io) for further assistance. We appreciate your support and understanding.

**Comment by thegreatgunbantoad** [comment]
Why are there 2 version 1.0s with diffrent silk screens?  The one I have has the version number on the other side.

**Reply by nfs0619** [reply]
Thanks for bringing this up!  
Could you please share a photo of your board so we can better understand the version you have?  
If you need more detailed support, you can also contact our technical support team at **[techsupport@seeed.io](mailto:techsupport@seeed.io)**. They’ll be happy to assist you.

**Comment by eisenrah** [comment]
Just a note for **Project 2: Detecting Temperature and Humidity** and **Option 2: DHT20 and LCD1602**...

*There seems to be an error in the steps:*
The Grove Temperature & Humidity Sensor V2.0 (DHT20) has to be connected to `I2C0`, not `D18`.
Then I connect the DHT20 sensor to `D18`, it cannot be initialized.

Looks like a copy and paste error from option 1 ;-)

**Reply by nfs0619** [reply]
Thank you for your feedback! We’ll check and fix the issue right away.

**Reply by ashleycawley** [reply]
eisenrah above is absolutely correct, using the code and advice supplied on the page D18 is the wrong port to connect the DHT20 to, connecting it instead to I2C0 works correctly. In addition to that there is a second mistake the guide advises to save the file named DHT20.py on the device where actually it needs to be lowercase: dht20.py the case sensitivity matters in this case and if it is capitalised on the Pico it will not work.

These issue took me quite some time to troubleshoot and overcome, the guide should be corrected in order to avoid frustration and problems with users just getting started with this kit.

**Reply by FaiyuetCik** [reply]
Thanks for the thorough investigation! Your feedback is really valuable — we've updated the Wiki accordingly. Thanks for helping improve the documentation!

**Comment by lugdunum-1964** [comment]
I went to these demos one by one. Most ran as expected, though I had to swap the ports on some of them. But I had issues with 2 of them:

Maybe I don't understand what the sound sensor is supposed to do. But It gave me the same values, whether or not there was sound. The sensor sat on my desk next to a speaker. With radio playing it gave me random values between 0 and 14000, which I would expect since the music is variable. But with the radio off it gave me the same values, while I was expecting (near) zeros, since there was silence. 

So I connected the buzzer from demo 1 and  just the sound sensor and used normal console prints to show the sensor value. With buzzer off, I expected near constant low values, with buzzer at +/- 1000Hz I'd expect some near constant higher values. But in both cases the value was random between 0 and 14000.

With the Pir the animation shows the fan and servo moving when waving a hand. 
But in my case they started moving immediately. So I replaced the servo and fan by a led, which switched on immediately again. It only went out when I moved something like 3-4m away from the sensor. And then on again when I came within 2-3m. Maybe that's expected (the specs on the wiki give a range of 3-6m. See https://www.seeedstudio.com/Grove-mini-PIR-motion-sensor-p-2930.html) But in that case the demo is just wrong for this type of sensor.

--
regards, Lyon

**Reply by Lukilyy** [reply]
Hi Lyon! Thanks for the detailed description. The behavior you're seeing is quite common, here's an explanation:
1. Regarding the Sound Sensor: This sensor is an analog microphone that outputs a real-time, instantaneous sound pressure signal. Even in a very quiet environment, there's inherent electronic noise (background noise) in the circuit, which causes the readings to fluctuate randomly within a range (like the 0-14000 you observed), rather than staying steadily at "0". When music or a buzzer is present, the complex and varying nature of the sound also results in random readings. This is the normal operating characteristic of this type of sensor.
2. Regarding the PIR motion sensor: We are extremely grateful for the valuable feedback you provided regarding this sensor. This is of great significance to us, and we will carefully record it and pay attention to this issue in the subsequent content maintenance, so as to provide users with more accurate references.
